### PR TITLE
Add CI guard for TypeScript codegen drift

### LINKED ISSUE #257: Add a CI guard for TypeScript codegen drift
## Summary

`docs/backend.md` ("Standalone TypeScript codegen") notes that the generated frontend API client is byte-identical to the committed output, "so CI can run the command and assert a clean `git diff` to catch api model/client drift." That guard is **not actually wired into CI** — `.github/workflows/run-tests.yml` never runs codegen.

Today, the only thing keeping `UI/src/lib/api/types` in sync with the backend DTOs/controllers is the author remembering to regenerate. A backend DTO change merged without regenerating would land stale client types with no automated signal. (I verified the committed output is currently in sync — running `dotnet run --project Game.Api -- codegen` produces no diff — so adding the guard now starts from green.)

## Suggested fix

Add a step to the `test-backend` job (it already has the .NET SDK set up):

```yaml
- name: Verify codegen is up to date
  run: |
    dotnet run --project Game.Api -- codegen
    git diff --exit-code -- UI/src/lib/api/types
```

This fails the build if a DTO/controller change wasn't accompanied by regenerated types.

### PR DESCRIPTION
**Pull request by ginderjeremiah**
## Summary

Wires the documented codegen-drift guard into CI. `docs/infrastructure.md` ("Standalone TypeScript codegen") states the generated frontend API client is byte-identical to the committed output "so CI can run the command and assert a clean `git diff` to catch api model/client drift" — but that guard was never actually in `.github/workflows/run-tests.yml`. Until now, the only thing keeping `UI/src/lib/api/types` in sync with the backend DTOs/controllers was the author remembering to regenerate.

## Change

Added a **Verify TypeScript codegen is up to date** step to the `test-backend` job (which already has the .NET SDK and a built solution):

```yaml
- name: Verify TypeScript codegen is up to date
  run: |
    dotnet run --project Game.Api --no-build --no-restore -- codegen
    git diff --exit-code -- UI/src/lib/api/types
```

A backend DTO/controller change merged without regenerating the client now fails this step instead of silently landing stale types.

## Notes / decisions

- **Placed in `test-backend`, right after `Build solution`.** It only needs the API assembly, gives fast feedback before the full suite, and the codegen command short-circuits before host construction (no Postgres/Redis/DI), so it's safe in CI.
- **`--no-build --no-restore`** rather than the issue's bare `dotnet run`, to reuse the binaries from the `Build solution` step — mirroring the existing `dotnet test --no-build --no-restore` convention in the same job. (Default config is Debug for both `dotnet build` and `dotnet run`, so the reuse is valid.)
- **Diff scoped to `UI/src/lib/api/types`** as the issue suggests — that's the only directory the generator writes to.
- **Known gap (not expanding scope here):** the `test-backend` job only runs when a backend path changes (per the `detect-changes` filter), so a hand-edit of the generated types under a *frontend-only* change wouldn't trigger this guard. That's an edge case (the types are generated, not hand-authored) and orthogonal to the issue's goal of catching forgotten regeneration after backend changes; happy to file a follow-up if you'd like it covered.

## Test plan

- [x] `dotnet build Game.Api` — clean, 0 warnings.
- [x] `dotnet run --project Game.Api --no-build --no-restore -- codegen` then `git diff --exit-code -- UI/src/lib/api/types` — **no diff** (starts from green, confirming the committed output is currently in sync, exactly as the new CI step will run it).
- [x] Workflow YAML validated (`yaml.safe_load`).

Closes #257

---
_Generated by [Claude Code](https://claude.ai/code/session_011cyhDdfZtZMMddM9KW9vpr)_

### CODE REVIEW

**Comment by ginderjeremiah** [comment]
Note for reviewers: CI on this PR is green but the `test-backend` job (where the new **Verify TypeScript codegen is up to date** step lives) was **skipped** — the only changed file is `.github/workflows/run-tests.yml`, which isn't in the `detect-changes` backend/frontend path filters, so all three test jobs are path-filtered out. The new step will therefore first execute on the next PR that touches a backend or frontend path. I verified it locally on this branch: `dotnet run --project Game.Api --no-build --no-restore -- codegen` followed by `git diff --exit-code -- UI/src/lib/api/types` exits clean (no drift).

Possible follow-up (left out to keep this PR scoped): add `.github/workflows/**` to the `detect-changes` filters so workflow changes self-validate against the test jobs. Happy to file that as a separate issue if you'd like it.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -78,8 +78,18 @@ jobs:
...
       run: dotnet build
 
+    - name: Verify TypeScript codegen is up to date
+      # The generated frontend API client (UI/src/lib/api/types) must stay byte-identical to what the
+      # backend DTOs/controllers produce. Regenerate and assert a clean diff so a backend change that
+      # forgets to regenerate fails here instead of landing stale client types. The codegen command
+      # short-circuits before host construction (no Postgres/Redis/DI), so it's safe in CI; reuse the
+      # binaries from "Build solution" (--no-build --no-restore), mirroring the test step below.
+      run: |
+        dotnet run --project Game.Api --no-build --no-restore -- codegen
```
Comment (ginderjeremiah) [inline .github/workflows/run-tests.yml:88]: Nit / follow-up (non-blocking): this step lives in `test-backend`, which only runs when `detect-changes.outputs.backend == 'true'` (filter at lines 33-47, which does **not** include `.github/**`). So the guard covers the issue's target scenario — a backend DTO/controller change that forgot to regenerate — but skips two cases: (1) a frontend-only change that hand-edits the generated types (you already called this out as out-of-scope), and (2) a workflow-only change like this very PR, where `test-backend` was `skipped` and the new step never executed in CI. Both are inherent to the path-filtered job design and don't undermine the issue's goal, so I wouldn't block on them — but a follow-up issue to run the codegen guard whenever `UI/src/lib/api/types/**` (or the workflow itself) changes would close the loop. The placement, `--no-build --no-restore` reuse, and scoped `git diff` are all correct and match the documented contract in `docs/infrastructure.md`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by ginderjeremiah — COMMENTED** [review]
## Review: Wire the documented TypeScript codegen-drift guard into CI (#257)

**Verdict:** Approve (posted as a comment because I'm the PR author, so GitHub won't accept a gating event on my own PR) — the change correctly implements the issue's suggested fix, and the technical claims hold up against the code and docs.

The new **Verify TypeScript codegen is up to date** step regenerates the client and asserts a clean `git diff` scoped to `UI/src/lib/api/types`, exactly the contract documented in `docs/infrastructure.md` ("Standalone TypeScript codegen", line 18). I verified the supporting code: `CodeGenCommand` is intercepted in `Startup.Main` before host construction, and `CodeGenPaths` writes only to `UI/src/lib/api/types` — so the command is safe in CI (no Postgres/Redis/DI) and the diff scope is right. Placement after `Build solution` with `--no-build --no-restore` correctly reuses the Debug binaries and mirrors the existing `dotnet test --no-build --no-restore` convention in the same job.

### Should-fix
- _None._

### Nits / follow-up
- The guard lives in `test-backend`, which is path-filtered to backend changes only (`.github/**` is not in the `backend` filter). It therefore covers the issue's target scenario (backend DTO/controller change without regeneration) but skips a frontend-only hand-edit of the generated types (you flagged this) and a workflow-only change like this PR — where `test-backend` was `skipped`, so the new step did not self-validate in this PR's own CI. Both are inherent to the path-filtered design and don't block; worth a follow-up issue to also trigger the guard on `UI/src/lib/api/types/**` (and workflow) changes. (Inline note on the step.)

### Test coverage
- No app/domain code changed; this is a CI workflow addition, so the project's unit/integration/e2e rules don't apply. The PR's own test plan (build clean, `codegen` produces no diff from green, YAML validated) is the appropriate validation here. Note that because `test-backend` was skipped on this workflow-only PR, the step itself was not exercised end-to-end in CI — a backend-touching PR will be its first live run.

### What's good
- Minimal, well-scoped, and faithful to the issue: one step, no over-engineering. The inline comment explaining the short-circuit, the `--no-build --no-restore` reuse, and the diff scope is genuinely useful for the next maintainer, and the author proactively documented the known coverage gap rather than hiding it.

---
_Generated by [Claude Code](https://claude.ai/code)_
